lib/pbench/agent/modules/metaclass.py

```diff
@@ -0,0 +1,106 @@
+from pathlib import Path
+import json
+import os
+
+
+class BadConstruction(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return repr(self.msg)
+
+
+class ToolMetadata:
+    def __init__(self, mode, context, logger):
+        self.logger = logger
+        assert mode in (
+            "redis",
+            "json",
+        ), f"Logic bomb! Unexpected mode, {mode}, encountered constructing tool meta data"
+        if not context:
+            raise (BadConstruction("No proper context given"))
```

Comment:
Perhaps change to `assert context, f"Logic bomb!  No context given on ToolMetadata object construction"`?